ctam/interfaces/fw_update_ifc.py

```diff
@@ -111,37 +111,44 @@ def ctam_fw_update_precheck(self, image_type="default"):
         status_message = ""
         # if all component is update not needed then return update not needed. If any one component is updatable return updatable
         self.ctam_get_fw_version(PostInstall=0)
+
+        # Getting all Component Ids with Version from PLDM Bundle JSON file
+        BundleComponentIdsAndVersions = self.PLDMComponentVersions(image_type=image_type)
         for element in self.PreInstallDetails:
-            # Check ONLY the ones which are part of HttpPushURITargets
+        #     # Check ONLY the ones which are part of HttpPushURITargets
             if (
                 self.included_targets == []
                 or element["@odata.id"] in self.included_targets
             ):
-                msg = f"Pre Install Details: {element['Id']} : {element['SoftwareId']} : {element['Version']} : "
+                msg = f"Pre Install Details: {element['Id']} : {element['SoftwareId']} : {element.get('Version', 'NA')} : "
```

Comment:
Isn't "Version" a mandatory field per OCP spec?

Reply:
Glad you pointed this out :)

There is a discussion, pending conclusion, in OCP FW update spec meeting, where FirmwareInventory was allowed to call out config information. There are still discussing if we should mandate Version field for those cases. Till then, this should help cater to designs which don't have Version. 

Let's revisit this topic for 1.1 spec.